packages/flutter/test/material/ink_well_test.dart

```diff
@@ -1330,4 +1333,253 @@ void main() {
+      expectTest(false);
+    }
+
+    // InkWell have a key
+    await test((Key? key, {bool onTapDownChangeWrap = false, bool onTapChangeWrap = false}) {
+      bool wrap = false;
+
+      return MaterialApp(
+        home: Scaffold(
+          body: Center(
+            child: StatefulBuilder(
+              builder: (BuildContext context, void Function(void Function()) setState) {
+                Future<void> changeWrap() async {
+                  await Future<void>.delayed(const Duration(milliseconds: 50));
+
+                  setState(() {
+                    wrap = !wrap;
+                  });
+                }
+
+                Widget child = InkWell(
+                  key: key,
+                  onTapDown: (_) {
+                    if (onTapDownChangeWrap)
+                      changeWrap();
+                  },
+                  onTap: () {
+                    if (onTapChangeWrap)
+                      changeWrap();
+                  },
+                  child: Container(
+                    height: 160,
+                    alignment: Alignment.center,
+                  ),
+                );
+
+                if (wrap) {
+                  child = Container(
+                    margin: const EdgeInsets.only(top: 320),
+                    child: child,
+                  );
+                }
+
+                return child;
+              },
+            ),
+          ),
+        ),
+      );
+    });
+
+    // Ancestor widget have Key
+    await test((Key? key, {bool onTapDownChangeWrap = false, bool onTapChangeWrap = false}) {
+      bool wrap = false;
+
+      return MaterialApp(
+        home: Scaffold(
+          body: Center(
+            child: StatefulBuilder(
+              builder: (BuildContext context, void Function(void Function()) setState) {
+                Future<void> changeWrap() async {
+                  await Future<void>.delayed(const Duration(milliseconds: 50));
+
+                  setState(() {
+                    wrap = !wrap;
+                  });
+                }
+
+                Widget child = InkWell(
+                  onTapDown: (_) {
+                    if (onTapDownChangeWrap)
+                      changeWrap();
+                  },
+                  onTap: () {
+                    if (onTapChangeWrap)
+                      changeWrap();
+                  },
+                  child: Container(
+                    height: 160,
+                    alignment: Alignment.center,
+                  ),
+                );
+
+                child = Container(
+                  key: key,
+                  child: child,
+                );
+
+                if (wrap) {
+                  child = Container(
+                    margin: const EdgeInsets.only(top: 320),
+                    child: child,
+                  );
+                }
+
+                return child;
+              },
+            ),
+          ),
+        ),
+      );
+    });
+  });
```

Comment:
fwiw this test is written in a very confusing way (calling a closure with a closure and so on). Try printing out the code and then drawing lines showing the code flow -- it jumps up and down and up and down a bunch of times.

It would be clearer if there were no closures, maybe by making a stateful widget that wraps the behaviour you're testing (what is currently in the closure you pass to `test`, except using State instead of a StatefulBuilder).